docs: how to data type (#6)
meowdding/catharsis: docs/item_models/data_types.md

# Data Types

> TODO: which ones can be Conditional, Select and Range?????

Reviewer: idk if can make comments in here maybe with but this should be in one, maybe with `<!-- -->` ?

This page lists all available data types.

All Data Types can be found [here](https://github.com/SkyblockAPI/SkyblockAPI/blob/2.0/src/common/main/kotlin/tech/thatgravyboat/skyblockapi/api/datatype/DataTypes.kt).

IDs of Data Types can be found at these locations:
[GenericDataTypes](https://github.com/SkyblockAPI/SkyblockAPI/blob/2.0/src/common/main/kotlin/tech/thatgravyboat/skyblockapi/api/datatype/defaults/GenericDataTypes.kt),
[LoreDataTypes](https://github.com/SkyblockAPI/SkyblockAPI/blob/2.0/src/common/main/kotlin/tech/thatgravyboat/skyblockapi/api/datatype/defaults/LoreDataTypes.kt),
[MiningDataTypes](https://github.com/SkyblockAPI/SkyblockAPI/blob/2.0/src/common/main/kotlin/tech/thatgravyboat/skyblockapi/api/datatype/defaults/MiningDataTypes.kt),
[PersonalAccessoryDataTypes](https://github.com/SkyblockAPI/SkyblockAPI/blob/2.0/src/common/main/kotlin/tech/thatgravyboat/skyblockapi/api/datatype/defaults/PersonalAccessoryDataTypes.kt).

## Available Number Data Types

> Any Data Types that return any form of number (so Integer, Double, Float, Long, Short, Byte) are available to you.

Reviewer: ```suggestion
Data Types that return any form of number (so Integer, Double, Float, Long, Short, Byte, or Enums) are available by default.
```

### Custom Number Data Types

| Data Type              | Description                                            |
|------------------------|--------------------------------------------------------|
| Snowballs              | Returns the current snowballs in the snowball shooter. |
| Dungeonbreaker Charges | Returns the current charges.                           |
| Drill Fuel             | Returns the current drill fuel.                        |

## Available String Data Types

> Any Data Types that return a String are available to you.

Reviewer: ```suggestion
Data Types that return a String are available by default.
```

### Custom String Data Types

| Data Type | Description                                  |
|-----------|----------------------------------------------|
| Hook      | Returns the SkyBlockId of the applied hook   |
| Line      | Returns the lore line at the applied line    |
| Sinker    | Returns the SkyBlockId of the applied sinker |

## Available Boolean Data Types

> Any Data Types that return a Boolean are available to you.

Reviewer: ```suggestion
Data Types that return a Boolean are available by default.
```

## Unavailable Data Types

> Any Data Type that returns a complex object (so Lists, Maps, Pairs, or custom objects) are, unless custom handled above, unavailable to you.

Reviewer: ```suggestion
Any Data Type that returns a complex object (so Lists, Maps, Pairs, or custom objects) are unavailable, unless explicitly specified above.
```
